Bluetooth: Add runnable tests

Vudentz opened the pull request:
This adds runnable tests that covers UUID, L2CAP and GATT APIs which don't require extensive setup to be run.

Vudentz committed:
Bluetooth: tests: Add tests for GATT

This adds initial test for GATT APIs that don't require a connection
thus can be run without any extra setup.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: hci_prop_evt: Don't set NO_QEMU_SERIAL_BT_SERVER

It should no longer be needed to set NO_QEMU_SERIAL_BT_SERVER manually
when setting CONFIG_BT_NO_DRIVER.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: tests: Add tests for L2CAP

This adds initial test for L2CAP APIs that don't require a connection
thus can be run without any extra setup.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: tests: Add tests for UUID APIs

This adds initial test for UUID APIs.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz commented:
@jhedberg any feedback on this one?

jhedberg commented on tests/bluetooth/gatt/prj.conf:
```diff
@@ -0,0 +1,11 @@
+CONFIG_BT_DEBUG_LOG=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_GATT_DYNAMIC_DB=y
+
```
@Vudentz could you fix this trailing whitespace at EOF (the extra empty line), and then we could merge this.

Vudentz replied:
Fixed.

Vudentz committed:
cmake: Set NO_QEMU_SERIAL_BT_SERVER when CONFIG_BT_NO_DRIVER is set

When not driver is selected there should be no need to have a serial
attached.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
